projects/storefrontlib/src/cms-components/content/tab-paragraph-container/tab-paragraph-container.component.ts

```diff
@@ -1,41 +1,61 @@
-import { Component, ChangeDetectionStrategy } from '@angular/core';
-// import { AbstractProductComponent } from '../abstract-product-component';
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import {
+  CMSTabParagraphContainer,
+  CmsService,
+  WindowRef,
+} from '@spartacus/core';
+import { CmsComponentData } from '../../../cms-structure/page/model/index';
+import { map, take } from 'rxjs/operators';
 
 @Component({
   selector: 'cx-tab-paragraph-container',
   templateUrl: './tab-paragraph-container.component.html',
-  styleUrls: ['./tab-paragraph-container.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class TabParagraphContainerComponent {
-  // extends AbstractProductComponent {
-  // @ViewChild('tabContent') tabContent: Element;
-  // fetchData() {
-  //     // console.log(this.contextParameters.tab);
-  //     // this.contextParameters.tab.title = new BehaviorSubject<any>({});
-  //     // this.contextParameters.tab.title.next('test');
-  //     // this.contextParameters.tab.title.complete();
-  //     // const s = new AsyncSubject<any>();
-  //     // this.occProductSearchService.query(query)
-  //     //     .then((pageData) => {
-  //     //         s.next(pageData);
-  //     //         s.complete();
-  //     // });
-  //     // return s;
-  //     // if (!this.contextParameters.tab) {
-  //     //     this.contextParameters.tab = {};
-  //     // }
-  //     // this.contextParameters.tab.title = this.component.title;
-  //     // console.log(this.contextParameters.tab);
-  //     // this.cd.detectChanges();
-  //     // console.log(this.component.title);
-  //     // console.log(this.tabContent);
-  //     // console.log(this.contextParameters);
-  //     // this.contextParameters);
-  // }
-  // ngOnInit() {
-  //     this.contextParameters.tabTitle.next('test');
-  //     this.cd.detectChanges();
-  //     // super.fetchData();
-  // }
+  activatedElements: HTMLElement[] = [];
+
+  constructor(
+    public componentData: CmsComponentData<CMSTabParagraphContainer>,
+    private cmsService: CmsService,
+    private windowRef: WindowRef
+  ) {}
+
+  components$ = this.componentData.data$.pipe(
+    map((data: CMSTabParagraphContainer) => {
+      return data.components.split(' ').map(component =>
+        this.cmsService.getComponentData<any>(component).pipe(
+          map(tab => {
+            if (!tab.flexType) {
+              tab.flexType = tab.typeCode;
+            }
+            return { ...tab, title: `productTabs.${tab.uid}` };
+          })
+        )
+      );
+    }),
+    take(1)
+  );
+
+  select(event: MouseEvent, tab: HTMLElement): void {
```

Comment:
I might mis the point (and might have introduced earlier myself), but can't we simply do:

```
  activeTabNum = 0;
  select(tabNum: number): void {
    this.active = num;
  }
```

in combination with view logic:
```
<h3 [class.active]="i === active" (click)="select(i)">
      {{ component.title | cxTranslate }}
</h3>
```

Reply:
with this we can drop the lower level APIs and have a clean constructor

Reply:
Much better. I think I was so stuck on the old code that even after adding the number in the HTML I didn't see the possibility for simplification. 🤦‍♂